### PR TITLE
docs: better expect.extend docs

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/src/test-configuration-js.md [hunk 1]
```diff
@@ -207,3 +209,24 @@ test('amount', async () => {
...
+
+You can combine custom matchers from multiple files or modules.
+
+```js title="fixtures.ts"
+import { test, expect } from '@playwright/test';
+import { test as dbTest, expect as dbExpect } from 'database-test-utils';
+import { test as a11yTest, expect as a11yExpect } from 'a11y-test-utils';
+
+export const expect = expect.composedExpect([dbExpect, a11yExpect]);
+export const test = test.composedTest([dbTest, a11yTest]);
```
Comment (mxschmitt) [inline docs/src/test-configuration-js.md:223]:
```suggestion
export const expect = expect.composedExpect(dbExpect, a11yExpect);
export const test = test.composedTest(dbTest, a11yTest);
```

see https://github.com/microsoft/playwright/blob/e8523abea5ba64db71837acaad099604e502a88c/tests/playwright-test/expect.spec.ts#L901.

File: docs/src/test-configuration-js.md [hunk 2]
```diff
@@ -174,12 +174,14 @@ export const expect = baseExpect.extend({
     }
 
     const message = pass
-      ? () => this.utils.matcherHint('toHaveAmount', locator, expected, { isNot: this.isNot }) +
+      ? () => this.utils.matcherHint('toHaveAmount', undefined, undefined, { isNot: this.isNot }) +
```
Comment (mxschmitt) [inline docs/src/test-configuration-js.md:177]: I feel like its a good time to link to the official extend docs now so users know what `this.*` is.

https://jestjs.io/docs/expect#expectextendmatchers

File: docs/src/test-configuration-js.md [hunk 3]
```diff
@@ -207,3 +209,24 @@ test('amount', async () => {
...
+### Combine custom matchers from multiple modules
+
+You can combine custom matchers from multiple files or modules.
+
+```js title="fixtures.ts"
+import { test, expect } from '@playwright/test';
+import { test as dbTest, expect as dbExpect } from 'database-test-utils';
+import { test as a11yTest, expect as a11yExpect } from 'a11y-test-utils';
+
+export const expect = expect.composedExpect(dbExpect, a11yExpect);
```
Comment (dgozman) [inline docs/src/test-configuration-js.md:222]:
```suggestion
export const expect = composedExpect(dbExpect, a11yExpect);
```

File: docs/src/test-configuration-js.md [hunk 4]
```diff
@@ -207,3 +209,24 @@ test('amount', async () => {
...
+export const test = test.composedTest(dbTest, a11yTest);
```
Comment (dgozman) [inline docs/src/test-configuration-js.md:223]:
```suggestion
export const test = composedTest(dbTest, a11yTest);
```

File: docs/src/test-configuration-js.md [hunk 5]
```diff
@@ -207,3 +209,24 @@ test('amount', async () => {
 :::note
 Do not confuse Playwright's `expect` with the [`expect` library](https://jestjs.io/docs/expect). The latter is not fully integrated with Playwright test runner, so make sure to use Playwright's own `expect`.
 :::
+
+### Combine custom matchers from multiple modules
+
+You can combine custom matchers from multiple files or modules.
+
+```js title="fixtures.ts"
+import { test, expect } from '@playwright/test';
```
Comment (dgozman) [inline docs/src/test-configuration-js.md:218]:
```suggestion
import { composedTest, composedExpect } from '@playwright/test';
```

**Review by dgozman — APPROVED** [review]

File: docs/src/test-configuration-js.md (hunk 2 above)
Comment (pavelfeldman) [reply docs/src/test-configuration-js.md:177]: I'm not sure if we are ready to link external docs.

**Pull request merged**